### PR TITLE
API 800,1000 and 1200 support for StorageVolumes, StorageVolumeTemplates, StorageSystems, StoragePool

### PR DESCRIPTION
**Pull request by chebroluharika**
**Description**

API 800,1000 and 1200 support for StorageVolumes, StorageVolumeTemplates, StorageSystems, StoragePool

**Issues Resolved**
NA

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [x] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,10 @@
 - This release supports OneView Rest API versions 800/1000/1200 minimally where we can use OneView v4.10/v4.20/v5.0 with this SDK. No new fields are added/deleted to support OneView Rest API 800/1000/1200. Complete support will be done in next releases.
 - Extending support to API 600 with the following HPE OneView resource:
 - oneview_scope
+- oneview_volume_template
+- oneview_volume
```
Comment (AnushaY1916) [inline CHANGELOG.md]: Add resources below not here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,10 @@
...
+- oneview_storage_pool
+- oneview_storage_system
 
 ### Oneview Features supported
 - oneview_connection_template
```
Comment (AnushaY1916) [inline CHANGELOG.md:8]: Add the supported resources here in alphabetical order.

File: endpoints-support.md [hunk 3]
```diff
@@ -54,6 +54,49 @@
 |<sub>/rest/fc-networks/{id}</sub>                                                        | PATCH    | :heavy_minus_sign:   | :white_check_mark:   | :white_check_mark:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   | :heavy_minus_sign:   |
 |<sub>/rest/fc-networks/{id}</sub>                                                        | PUT      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
 |<sub>/rest/fc-networks/{id}</sub>                                                        | DELETE   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
+|     **Storage Volume Templates**
```
Comment (AnushaY1916) [inline endpoints-support.md:97]: Please these resources in alphabetical order

File: examples/storage_pool.rb [hunk 4]
```diff
@@ -31,7 +31,8 @@
   storage_system @storage_system_ip
   data(
     description: "SSD Storage pool - CHEF",
-    isManaged: false
+    isManaged: false,
```
Comment (AnushaY1916) [inline examples/storage_pool.rb:34]: Update the license to 2020 at the top of this file.

**Review by AnushaY1916 — CHANGES_REQUESTED** [review]

File: examples/volume_api500.rb [hunk 5]
```diff
@@ -37,6 +37,7 @@
   data(store_serv_properties)
   storage_pool 'CPG-SSD-AO'
   storage_system 'ThreePAR-1'
+  action :create_if_missing
```
Comment (soodpr) [inline examples/volume_api500.rb:40]: you don't need to specify here explicitly..if you make this action default here:https://github.com/HewlettPackard/oneview-chef/blob/master/resources/volume.rb#L23

File: libraries/resource_providers/api1000/synergy/storage_pool_provider.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API1000
+    module Synergy
+      # StoragePool API500 C7000 provider
```
Comment (soodpr) [inline libraries/resource_providers/api1000/synergy/storage_pool_provider.rb]: update this comment to API1000 Synergy provider

File: libraries/resource_providers/api800/c7000/storage_pool_provider.rb [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API800
+    module C7000
+      # StoragePool API800 C7000 provider
+      class StoragePoolProvider < OneviewCookbook::API500::C7000::StoragePoolProvider
```
Comment (soodpr) [inline libraries/resource_providers/api800/c7000/storage_pool_provider.rb]: it should inherit from 600

File: libraries/resource_providers/api800/c7000/storage_system_provider.rb [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API800
+    module C7000
+      # StorageSystem API600 C7000 provider
```
Comment (soodpr) [inline libraries/resource_providers/api800/c7000/storage_system_provider.rb]: update the comment

File: libraries/resource_providers/api800/synergy/storage_system_provider.rb [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  module API800
+    module Synergy
+      # StorageSystem API600 Synergy provider
```
Comment (soodpr) [inline libraries/resource_providers/api800/synergy/storage_system_provider.rb]: update the comment

File: CHANGELOG.md (hunk 1 above)
Comment (chebroluharika) [reply CHANGELOG.md]: corrected

File: CHANGELOG.md (hunk 2 above)
Comment (chebroluharika) [reply CHANGELOG.md:8]: corrected

File: endpoints-support.md (hunk 3 above)
Comment (chebroluharika) [reply endpoints-support.md:97]: corrected

File: examples/storage_pool.rb (hunk 4 above)
Comment (chebroluharika) [reply examples/storage_pool.rb:34]: corrected

File: examples/volume_api500.rb (hunk 5 above)
Comment (chebroluharika) [reply examples/volume_api500.rb:40]: done

File: libraries/resource_providers/api1000/synergy/storage_pool_provider.rb (hunk 6 above)
Comment (chebroluharika) [reply libraries/resource_providers/api1000/synergy/storage_pool_provider.rb]: corrected

File: libraries/resource_providers/api800/c7000/storage_pool_provider.rb (hunk 7 above)
Comment (chebroluharika) [reply libraries/resource_providers/api800/c7000/storage_pool_provider.rb]: corrected

File: libraries/resource_providers/api800/c7000/storage_system_provider.rb (hunk 8 above)
Comment (chebroluharika) [reply libraries/resource_providers/api800/c7000/storage_system_provider.rb]: corrected

File: libraries/resource_providers/api800/synergy/storage_system_provider.rb (hunk 9 above)
Comment (chebroluharika) [reply libraries/resource_providers/api800/synergy/storage_system_provider.rb]: corrected

**Review by soodpr — APPROVED** [review]
LGTM 👍

File: endpoints-support.md [hunk 10]
```diff
@@ -69,15 +69,58 @@
...
+|<sub>/rest/network-sets/{id}</sub>                                                       | PUT      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:  | :white_check_mark:   | :white_check_mark:   |
+|<sub>/rest/network-sets/{id}</sub>                                                       | DELETE   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:  | :white_check_mark:   | :white_check_mark:   |
 |<sub>/rest/network-sets/{id}</sub>                                                       | PATCH    | :heavy_minus_sign: | :white_check_mark: | :white_check_mark: | :heavy_minus_sign: |:heavy_minus_sign: | :heavy_minus_sign: | :heavy_minus_sign: |
+|      **Storage Pools**
+|<sub>/rest/storage-pools</sub>                                                           | GET      | :white_check_mark:   | :white_check_mark: | :white_check_mark:
```
Comment (jyothisgm) [inline endpoints-support.md]: Please update the endpoint support for the resources till 1200.

File: endpoints-support.md [hunk 11]
```diff
@@ -69,15 +69,58 @@
...
 |<sub>/rest/uplink-sets/{id}</sub>                                                        | GET      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |
 |<sub>/rest/uplink-sets/{id}</sub>                                                        | PUT      | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |
 |<sub>/rest/uplink-sets/{id}</sub>                                                        | DELETE   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |:white_check_mark:   |:white_check_mark:   |:white_check_mark:   |
+|     **Volumes**
+|<sub>/rest/storage-volumes</sub>                                                         | GET      |  :white_check_mark:   | :white_check_mark:   | :white_check_mark:   |
```
Comment (jyothisgm) [inline endpoints-support.md]: Same as above comment.

**Review by AnushaY1916 — APPROVED** [review]

**Review by jyothisgm — CHANGES_REQUESTED** [review]
The branch as removed the previous commits. Please change it or close the PR. LE and Server Handware seems to be removed while fixing conflicts.

**Pull request closed without merging**